public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -66,6 +66,7 @@ export function GraphiteDatasource(this: any, instanceSettings, $q, backendSrv,
     const proxyMode = !this.url.match(/^http/);
     if (proxyMode) {
       httpOptions.headers['X-Dashboard-Id'] = options.dashboardId;
+      httpOptions.headers['X-Dashboard-UId'] = options.dashboardUId;
```

Comment:
I think we can do this more generically inside https://github.com/grafana/grafana/blob/master/public/app/core/services/backend_srv.ts#L151 . 

From there figure out what dashboard is open add these headers in one location for all data source requests